iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/RoleBasedAccessControlCheck.java

```diff
@@ -19,70 +19,61 @@
  */
 package org.sonar.iac.terraform.checks.azure;
 
+import org.sonar.api.utils.Version;
 import org.sonar.check.Rule;
-import org.sonar.iac.common.api.checks.CheckContext;
-import org.sonar.iac.common.checks.PropertyUtils;
-import org.sonar.iac.common.checks.TextUtils;
-import org.sonar.iac.terraform.api.tree.AttributeTree;
-import org.sonar.iac.terraform.api.tree.BlockTree;
-import org.sonar.iac.terraform.checks.AbstractResourceCheck;
+import org.sonar.iac.terraform.checks.AbstractNewResourceCheck;
+import org.sonar.iac.terraform.checks.utils.ExpressionPredicate;
+import org.sonar.iac.terraform.symbols.AttributeSymbol;
+import org.sonar.iac.terraform.symbols.BlockSymbol;
+
+import static org.sonar.iac.terraform.checks.utils.ExpressionPredicate.isFalse;
+import static org.sonar.iac.terraform.plugin.TerraformProviders.Provider.Identifier.AZURE;
 
 @Rule(key = "S6383")
-public class RoleBasedAccessControlCheck extends AbstractResourceCheck {
+public class RoleBasedAccessControlCheck extends AbstractNewResourceCheck {
 
   private static final String MISSING_MESSAGE = "Omitting '%s' disables role-based access control for this resource. Make sure it is safe here.";
   private static final String DISABLED_MESSAGE = "Make sure that disabling role-based access control is safe here.";
 
+  private static final Version AZURE_V_3 = Version.create(3, 0);
+
   @Override
-  protected void registerResourceChecks() {
-    register(RoleBasedAccessControlCheck::checkKubernetesCluster, "azurerm_kubernetes_cluster");
-    register(RoleBasedAccessControlCheck::checkKeyVault, "azurerm_key_vault");
-  }
+  protected void registerResourceConsumer() {
+    // Report if 'role_based_access_control' block is missing or check its attributes.
+    // Report if 'role_based_access_control.enabled' is set to false
+    // or check 'role_based_access_control.azure_active_directory' block
+    // Report if 'role_based_access_control.azure_active_directory.managed' is set to true
+    // and 'role_based_access_control.azure_active_directory.managed' is missing or set to false.
```

Comment:
```suggestion
    // and 'role_based_access_control.azure_active_directory.azure_rbac_enabled' is missing or set to false.
```
Not sure about that one